crates/biome_fs/src/path.rs

```diff
@@ -50,7 +125,254 @@ impl BiomePath {
         read_to_string(path)
     }
 
+    /// Returns the extension of the path
     pub fn extension_as_str(&self) -> Option<&str> {
         self.extension().and_then(OsStr::to_str)
     }
+
+    /// Returns the file name of the path
+    fn get_file_name(&self) -> Option<&OsStr> {
+        self.path.file_name()
+    }
+
+    /// The priority of the file.
+    /// - `biome.json` and `biome.jsonc` have the highest priority
+    /// - `package.json` and `tsconfig.json`/`jsconfig.json` have the second-highest priority, and they are considered as manifest files
+    /// - Other files are considered as files to handle
+    fn priority(file_name: &OsStr) -> FileKinds {
+        if file_name == ConfigName::biome_json() || file_name == ConfigName::biome_jsonc() {
+            FileKind::Config.into()
+        } else if matches!(
+            file_name.as_encoded_bytes(),
+            b"package.json" | b"tsconfig.json" | b"jsconfig.json"
+        ) {
+            FileKind::Manifest.into()
+        } else {
+            FileKind::ToHandle.into()
+        }
+    }
+
+    pub fn is_config(&self) -> bool {
+        self.kind.0.contains(FileKind::Config)
+    }
+
+    pub fn is_manifest(&self) -> bool {
+        self.kind.0.contains(FileKind::Manifest)
+    }
+
+    pub fn is_ignore(&self) -> bool {
+        self.kind.0.contains(FileKind::Ignore)
+    }
+
+    pub fn is_to_inspect(&self) -> bool {
+        self.kind.0.contains(FileKind::ToInspect)
+    }
+}
+
+#[cfg(feature = "serde")]
+impl schemars::JsonSchema for FileKinds {
+    fn schema_name() -> String {
+        String::from("FileKind")
+    }
+
+    fn json_schema(gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+        <Vec<FileKind>>::json_schema(gen)
+    }
+}
+
+impl Deref for BiomePath {
+    type Target = PathBuf;
+
+    fn deref(&self) -> &Self::Target {
+        &self.path
+    }
+}
+
+impl PartialOrd for BiomePath {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for BiomePath {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let current_file_name = self.get_file_name();
+        let other_file_name = other.get_file_name();
+        match (current_file_name, other_file_name) {
+            (Some(current_file_name), Some(other_file_name)) => {
+                match Self::priority(current_file_name).cmp(&Self::priority(other_file_name)) {
+                    Ordering::Equal => self.path.cmp(&other.path),
+                    ordering => ordering,
+                }
+            }
+            (Some(_), None) => Ordering::Less,
+            (None, Some(_)) => Ordering::Greater,
+            (None, None) => Ordering::Equal,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Hash, Default)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize, schemars::JsonSchema)
+)]
+/// It defines an alias for a path
```

Comment:
Convention is to put the docs before the annotations.

```suggestion
/// It defines an alias for a path
#[derive(Debug, Clone, Hash, Default)]
#[cfg_attr(
    feature = "serde",
    derive(serde::Serialize, serde::Deserialize, schemars::JsonSchema)
)]
```